File: 프로그래머스/Level2/소수 만들기.cpp
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

int answer = 0;

vector<int> numList;
int num[3];

bool primeNum(){
    
    int sum = num[0]+num[1]+num[2];
    
    for(int i=2; i<=sqrt(sum); i++){
        if(sum%i == 0) return false;
    }
    
    return true;
}

void dfs(int inx, int cnt){

    if(cnt == 3){
        if(primeNum()) answer ++;
        
        return;
    }
    
    for(int i=inx; i<numList.size(); i++){
        num[cnt] = numList[i];
        dfs(i+1, cnt+1);
    }
}

int solution(vector<int> nums) {
    numList = nums;
    dfs(0, 0);
    return answer;
}

File: 프로그래머스/Level1/소수 만들기.js
function solution(nums) {
    let answer = 0;
    
    for(let q=0; q<nums.length-2; q++)
        for(let w=q+1; w<nums.length-1; w++)
            for(let e=w+1; e<nums.length; e++)
                if(findPrimeNumber(nums[q], nums[w], nums[e])) answer ++;
    
    return answer;
}

function findPrimeNumber(n1, n2, n3){
    const sum = n1 + n2 + n3;
    
    for(let i=2; i<=Math.sqrt(sum); i++)
        if(sum%i === 0) return false;
        
    return true;
}
